### ISSUE TITLE
Design for UI / Frontend

### ISSUE THREAD

**Issue by a-jaxell** [fungover/mmotodo#4]
# Design of UI 

Since the only design requirements of the UI are :

* _"Backend och **frontend med single page application**"._
* _"Develop a task management system where users can **create, assign, and track tasks**"_

I think we should discuss what the whole thing could look like.  I broke out a few views that i think needs to be a part of the project in order to fulfil the design requirements. 

---

- [ ] #19
- [x] #21
- [ ] #50
- [ ] #18
- [ ] A view for unassigned Tasks
- [ ] A view of Teams and associated Users

---

What do you think? What else should be part of the UI?

**Comment by liber09** [comment]
> # Design of UI
> Since the only design requirements of the UI are :
> 
> * _"Backend och **frontend med single page application**"._
> * _"Develop a task management system where users can create, assign, and track tasks"_
> 
> I think we should discuss what the whole thing could look like. I broke out a few views that i think needs to be a part of the project in order to fulfil the design requirements.
> 
> * [ ]  A Login page
> * [ ]  A view Tasks associated with an User
> * [ ]  A view of Tasks associated with a team
> * [ ]  A view for creating a new Task
> 
> What do you think? What else should be part of the UI?

I agree on those views to fulfill the design requirements. Maybe a view of tasks that is not associated to a user or a team. Unassigned tasks view maybe?

**Comment by a-jaxell** [comment]
> I agree on those views to fulfill the design requirements. Maybe a view of tasks that is not associated to a user or a team. Unassigned tasks view maybe?

A view for Unassigned tasks is a good idea, It could be useful to bring attention to Tasks that are sort of "hanging" in the air.
I'll add this to the list above.

**Comment by a-jaxell** [comment]
> I think the login will be redirected to an auth service which will handle the login page under its own domain. So what we need is more a login/logout link or button?

A Login/Logout seems sensible in this case. Updating that.

> Some way of associating users/team with a task would be good too. Can be part of the flow for creating a new task but also something that can be done later in its own flow.

Initially I was thinking that Tasks should be created unassigned and then assignment would be after creation. But I think like you're saying that assignment upon creation could probably be better. 

Another thing I was thinking about is wether Tasks should be able to be unassigned both on Team level and User level. Should it be possible to create a Task that has no relation with an User or a Team?

As for vertical or horizontal slicing I think vertical would probably accelerate progress as soon as we've agreed on the database models.

**Comment by davidlidebrandt** [comment]
Perhaps it would be a good idea to also create an issue where we specify the colors, fonts etc. to create a coherent look across the whole application. Not sure if this has been brought up yet, but I think it would be a good idea, could also be extended to a full "design system" with specified margins, paddings and so on.

**Comment by kappsegla** [comment]
> Perhaps it would be a good idea to also create an issue where we specify the colors, fonts etc. to create a coherent look across the whole application. Not sure if this has been brought up yet, but I think it would be a good idea, could also be extended to a full "design system" with specified margins, paddings and so on.

Is this something that can be done as a default css file that styles the application and can be linked in the single page that will be the frontend? Make a branch and pull request with such a file or files?

**Comment by davidlidebrandt** [comment]
Yes, I think that some simple styling could be applied quite easily that way, and we can also provide classes and variables in the same file for additional use. If it sounds like a good idea I will create a separate issue for this as done for the other parts.

**Comment by 3xer** [comment]
I like the table design it's nice and easy to understand from #21. Have we come up with a homepage "default view" and what the whole page should look like? Couldn't find any designs for the site as a whole just the smaller components.
